src/Tools/Model/Permissionable.php

```diff
@@ -0,0 +1,39 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core\Tools\Model;
+
+use LotGD\Core\Models\Permission;
+
+/**
+ * Tools to work with a permission type field.
+ */
+trait Permissionable
+{
+    /** @var array Associations between permission-id and PermissionAssociation entity. */
+    private $_permissions = [];
+
+    protected function loadPermissions()
+    {
+        if (empty($this->_permissions)) {
+            foreach ($this->permissions as $permission) {
+                $this->_permissions[$permission->getId()] = $permission;
+            }
+        }
+    }
+
+    public function hasPermission(string $permissionId): bool
```

Comment:
`hasPermissionSet` is clearer that it's a presence flag. `hasPermission` sounds like "Do I have permission?" which sounds like "Is allowed?".